Library/src/main/java/com/shopify/testify/internal/compare/colorspace/DeltaE.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Shopify Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.shopify.testify.internal.compare.colorspace
+
+import kotlin.math.abs
+import kotlin.math.atan2
+import kotlin.math.cos
+import kotlin.math.exp
+import kotlin.math.pow
+import kotlin.math.sin
+import kotlin.math.sqrt
+
+/**
+ * The CIELAB color space (also known as CIE L*a*b* or sometimes abbreviated as simply "Lab" color
+ * space) is a color space defined by the International Commission on Illumination (CIE) in 1976.
+ * It expresses color as three values: L* for the lightness from black (0) to white (100), a* from
+ * green (−) to red (+), and b* from blue (−) to yellow (+). CIELAB was designed so that the same
+ * amount of numerical change in these values corresponds to roughly the same amount of visually
+ * perceived change.
+ *
+ * The CIE calls their distance metric ΔE*ab where delta is a Greek letter often used to denote
+ * difference, and E stands for Empfindung; German for "sensation".
+ *
+ * If deltaE < 1 then the difference can't recognized by human eyes.
+ *
+ * https://en.wikipedia.org/wiki/Color_difference#CIEDE2000
+ *
+ * Calculate the colour difference value between two colours in lab space.
+ *
+ * @param L1 first colour's L component
+ * @param a1 first colour's a component
+ * @param b1 first colour's b component
+ * @param L2 second colour's L component
+ * @param a2 second colour's a component
+ * @param b2 second colour's b component
+ * @return the CIE 2000 colour difference
+ */
+fun calculateDeltaE(L1: Double, a1: Double, b1: Double, L2: Double, a2: Double, b2: Double): Double {
```

Comment:
Is this not part of the lib?

Reply:
OK. It does not. I thought I was going crazy. I had found the deltaE conversion on wikipedia and was looking for some RGB --> LAB conversion code and found this `colormath` library. I'm only using the lib for colorspace conversions. I checked and they don't have any methods for difference or deltaE.

But, now that I see this, I wonder if the author of that library would find the `deltaE` calculation useful? It might be worth PR'ing that.